internal/services/authorization/azuresdkhacks/definitions.go

```diff
@@ -7,151 +7,119 @@ import (
+		return
 	}
 
 	return
 }
 
-// GetResponder handles the response to the Get request. The method always
-// closes the http.Response Body.
-func (client RoleDefinitionsWorkaroundClient) GetResponder(resp *http.Response) (result RoleDefinitionGetResponse, err error) {
-	err = autorest.Respond(
-		resp,
-		azure.WithErrorUnlessStatusCode(http.StatusOK),
-		autorest.ByUnmarshallingJSON(&result),
-		autorest.ByClosing())
-	result.Response = autorest.Response{Response: resp}
-	return
+type CreateOrUpdateOperationResponse struct {
+	HttpResponse *http.Response
+	OData        *odata.OData
+	Model        *RoleDefinition
 }
 
-// RoleDefinition role definition.
-type RoleDefinitionGetResponse struct {
-	autorest.Response `json:"-"`
-	// ID - READ-ONLY; The role definition ID.
-	ID *string `json:"id,omitempty"`
-	// Name - READ-ONLY; The role definition name.
-	Name *string `json:"name,omitempty"`
-	// Type - READ-ONLY; The role definition type.
-	Type *string `json:"type,omitempty"`
-	// RoleDefinitionProperties - Role definition properties.
-	*RoleDefinitionProperties `json:"properties,omitempty"`
+type GetOperationResponse struct {
+	HttpResponse *http.Response
+	OData        *odata.OData
+	Model        *RoleDefinition
 }
 
-type RoleDefinitionUpdateResponse struct {
-	autorest.Response `json:"-"`
-	// ID - READ-ONLY; The role definition ID.
-	ID *string `json:"id,omitempty"`
-	// Name - READ-ONLY; The role definition name.
-	Name *string `json:"name,omitempty"`
-	// Type - READ-ONLY; The role definition type.
-	Type *string `json:"type,omitempty"`
-	// RoleDefinitionProperties - Role definition properties.
-	*RoleDefinitionProperties `json:"properties,omitempty"`
+type RoleDefinition struct {
+	Id         *string                   `json:"id,omitempty"`
+	Name       *string                   `json:"name,omitempty"`
+	Properties *RoleDefinitionProperties `json:"properties,omitempty"`
+	Type       *string                   `json:"type,omitempty"`
 }
 
-// RoleDefinitionProperties role definition properties.
 type RoleDefinitionProperties struct {
-	// RoleName - The role name.
-	RoleName *string `json:"roleName,omitempty"`
-	// Description - The role definition description.
-	Description *string `json:"description,omitempty"`
-	// RoleType - The role type.
-	RoleType *string `json:"type,omitempty"`
-	// Permissions - Role definition permissions.
-	Permissions *[]Permission `json:"permissions,omitempty"`
-	// AssignableScopes - Role definition assignable scopes.
-	AssignableScopes *[]string `json:"assignableScopes,omitempty"`
-
+	AssignableScopes *[]string                     `json:"assignableScopes,omitempty"`
+	Description      *string                       `json:"description,omitempty"`
+	Permissions      *[]roledefinitions.Permission `json:"permissions,omitempty"`
+	RoleName         *string                       `json:"roleName,omitempty"`
+	Type             *string                       `json:"type,omitempty"`
 	// not exposed in the sdk
```

Comment:
These fields are available [in the newer API version](https://github.com/hashicorp/go-azure-sdk/blob/5d2118ecfed532e1549552249c29398aea44653d/resource-manager/authorization/2022-04-01/roledefinitions/model_roledefinitionproperties.go#L14-L21) which is [already used by the Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/0d48f0d6f542d5a5b780e63b4e7d446660c69bfe/internal/services/authorization/client/client.go#L27) - as such I believe this workaround is no longer needed?

Reply:
I planed to remove them in another PR, and do the API version upgrade at that PR, how do you think?